core/src/transport/memory.rs

```diff
@@ -184,15 +181,12 @@ impl Transport for MemoryTransport {
         id: ListenerId,
         addr: Multiaddr,
     ) -> Result<(), TransportError<Self::Error>> {
-        let port = if let Ok(port) = parse_memory_addr(&addr) {
-            port
-        } else {
+        let Ok(port) = parse_memory_addr(&addr) else {
```

Comment:
This could just be `unwrap_or()?`.

Reply:
I tried something like this out:

```rust
let port = parse_memory_addr(&addr).unwrap_or(return Err(TransportError::MultiaddrNotSupported(addr)))?;
```

but this causes compilation issues. I tried `unwrap_or_else` as well, but any return statement inside the `unwrap`  tries to return a value to the `port` variable. Maybe I'm missing something here

Reply:
Remove the `return` and the wrapping in `Err` and it should work :)

Reply:
Actually if it is already a `Result`, you might need to use `map_err`. `unwrap_or` is for `Option`s.

Reply:
ah yes, looks like `map_err` is the right one

```rust
let port = parse_memory_addr(&addr).map_err(|_| TransportError::MultiaddrNotSupported(addr))?;
```

 thanks!